File: imaipay/functions/fix_imports.py
import os
import glob
import re

files = glob.glob('src/**/*.ts', recursive=True)
for file in files:
    with open(file, 'r') as f:
        content = f.read()

    uses_timestamp = 'Timestamp' in content
    uses_fieldvalue = 'FieldValue' in content

    if uses_timestamp or uses_fieldvalue:
        imports = []
        if uses_timestamp: imports.append('Timestamp')
        if uses_fieldvalue: imports.append('FieldValue')

        if 'from "firebase-admin/firestore"' in content:
            # Add to existing import
            def repl(m):
                existing = [x.strip() for x in m.group(1).split(',')]
                for i in imports:
                    if i not in existing:
                        existing.append(i)
                return f'import {{ {", ".join(existing)} }} from "firebase-admin/firestore";'
            
            content = re.sub(r'import\s*\{\s*([^}]+)\s*\}\s*from\s*"firebase-admin/firestore";', repl, content)
        else:
            # Add new import at top
            content = f'import {{ {", ".join(imports)} }} from "firebase-admin/firestore";\n' + content

    with open(file, 'w') as f:
        f.write(content)

File: imaipay/functions/fix_imports.js
const fs = require('fs');
const glob = require('glob');

const files = glob.sync('src/**/*.ts');
for (const file of files) {
  let content = fs.readFileSync(file, 'utf8');
  
  const usesTimestamp = content.includes('Timestamp');
  const usesFieldValue = content.includes('FieldValue');
  
  if (usesTimestamp || usesFieldValue) {
    const imports = [];
    if (usesTimestamp) imports.push('Timestamp');
    if (usesFieldValue) imports.push('FieldValue');
    
    // Check if there is already an import from "firebase-admin/firestore"
    if (content.includes('from "firebase-admin/firestore"')) {
      // Just replace it. This is a bit hacky but we know wallet.ts has `import { Transaction } from "firebase-admin/firestore";`
      content = content.replace(/import\s*{([^}]+)}\s*from\s*"firebase-admin\/firestore";/, (match, p1) => {
         const existing = p1.split(',').map(s => s.trim()).filter(Boolean);
         imports.forEach(i => { if (!existing.includes(i)) existing.push(i); });
         return `import { ${existing.join(', ')} } from "firebase-admin/firestore";`;
      });
    } else {
      content = `import { ${imports.join(', ')} } from "firebase-admin/firestore";\n` + content;
    }
    
    fs.writeFileSync(file, content);
  }
}
